### PR TITLE
mock-pop-mining: BitcoinBlockData: implement the Merkle tree functionality; add tests

### PR DESCRIPTION
**Pull request by eu321**

### CODE REVIEW

File: mock-pop-mining/src/test/java/org/veriblock/sdk/mock/BitcoinBlockchainTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+public class BitcoinBlockchainTest {
+
+    // FIXME: need a proper genesis block
+    private final byte[] rawGenesis = Utils.decodeHex("00000020e1205431708b69b0a0c48ef51d3bdd1049643cfa7e1d841a18d43dcf48d6be5ef3bb206027319c0939db20cf223be3fc7385d1aa871461ce89a441577978f40afd9adf5dffff7f2001000000");
+    private final BitcoinBlock genesis = SerializeDeserializeService.parseBitcoinBlock(rawGenesis);
+        
+    private final BitcoinBlockchainBootstrapConfig bootstrap = new BitcoinBlockchainBootstrapConfig(
+                                                                    Arrays.asList(genesis), 92);
+
```
Comment (blackyblack) [inline mock-pop-mining/src/test/java/org/veriblock/sdk/mock/BitcoinBlockchainTest.java:40]: Why 92?

**Review by blackyblack — APPROVED** [review]

File: mock-pop-mining/src/test/java/org/veriblock/sdk/mock/BitcoinBlockchainTest.java (hunk 1 above)
Comment (eu321) [reply mock-pop-mining/src/test/java/org/veriblock/sdk/mock/BitcoinBlockchainTest.java:40]: block taken from the regtest dataset

Comment (blackyblack) [reply mock-pop-mining/src/test/java/org/veriblock/sdk/mock/BitcoinBlockchainTest.java:40]: Could we make it self-describing? Maybe define new constant with some meaningful name.

Comment (eu321) [reply mock-pop-mining/src/test/java/org/veriblock/sdk/mock/BitcoinBlockchainTest.java:40]: this is just a temporary block that will be replaced with proper genesis block

**Review by blackyblack — APPROVED** [review]

**Pull request merged**